src/raytracer.rs

```diff
@@ -1,90 +1,76 @@
-extern crate num_cpus;
-extern crate crossbeam_utils;
-
-use na::Vector3 as Vec3;
-use std::f64::INFINITY;
 use camera::Camera;
 use light::Light;
 use object::Object;
-use std::sync::Arc;
+use surface::{Division, Surface};
+
+use na::Vector3 as Vec3;
+use scoped_threadpool::Pool;
+use num_cpus;
+
+use std::f64::INFINITY;
 
 pub struct Ray {
     pub dir: Vec3<f64>,
     pub origin: Vec3<f64>
 }
 
 pub fn march (cam: &Camera, objects: &Vec<Box<Object>>, lights: &Vec<Box<Light>>)
```

Comment:
`&Vec<Box<Object>>` can be converted to `&[Box<Object>]`, that'd make it more universal. Same thing for `lights`

Reply:
I'll try that right now.